File: Volume-1-(100-199)/The-3n+1-problem/answer.c
#include <stdio.h>
#include "Math.h"
int cycle_length(int n){
    int l = 1;
    while (n != 1){
        if (n % 2 != 0){
            n = 3 * n + 1;
        } else {
            n = n / 2;
        }
        l++;
    }
    return l;
}

int max_cycle_length(int i, int j){
    int max = 0;
    int k;
    for (k = Math_Min(i, j); k <= Math_Max(i, j); k++){
        max = Math_Max(max, cycle_length(k));
    }
    return max;
}

int main() {
    int i, j;
     while (scanf("%d %d", &i, &j) == 2) {
        int maxLength = max_cycle_length(i, j);
        printf("%d %d %d\n", i, j, maxLength);
    }
    return 0;
}

File: Problem-Set-Volumes-(100...1999)/Volume-1-(100-199)/100 - The 3n + 1 problem/answer.js
const { createInterface } = require("readline/promises");
const { stdin: input, stdout: output } = require("node:process");
function cycleLength(n) {
  var l = 1;
  while (n !== 1) {
    if (n % 2 !== 0) {
      n = 3 * n + 1;
    } else {
      n = n / 2;
    }
    l++;
  }
  return l;
}
function maxCycleLength(i, j) {
  var max = 0;
  for (let k = Math.min(i, j); k <= Math.max(i, j); k++) {
    max = Math.max(max, cycleLength(k));
  }
  return max;
}
(async () => {
  const interface = createInterface({ input, output });
  while (true) {
    const input = await interface.question("Enter i and j or e for exit: ");
    if (input.toLocaleLowerCase() === "e") break;
    const [i, j] = input.split(" ").map(Number);
    const maxLength = maxCycleLength(i, j);
    console.log(`${i} ${j} ${maxLength}`);
  }
  interface.close();
})();
